samples/js/flickr/slurp/src/flickr.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import flickrAPI from 'flickr-oauth-and-upload';
+import readline from 'readline';
+import {
+  invariant,
+  jsonToNoms,
+  Struct,
+} from '@attic/noms';
+
+export default class Flickr {
+  apiKey: string;
+  apiSecret: string;
+  accessToken: string;
+  accessTokenSecret: string;
+
+  constructor(apiKey: string, apiSecret: string,
+              accessToken: string = '', accessTokenSecret: string = '') {
+    this.apiKey = apiKey;
+    this.apiSecret = apiSecret;
+    this.accessToken = accessToken;
+    this.accessTokenSecret = accessTokenSecret;
+  }
+
+  async authenticate(): Promise<void> {
+    const [token, secret, url] = await getAuthToken(this.apiKey, this.apiSecret);
+    const verificationCode = await promptForVerificationCode(url);
+    // $FlowIssue: Flow does not understand destructuring assignment.
+    [this.accessToken, this.accessTokenSecret] =
+        await this.getAccessToken(token, secret, verificationCode);
+  }
+
+  getAccessToken(oauthToken: string, oauthTokenSecret: string, oauthVerifier: string):
+      Promise<[string, string]> {
+    return new Promise((resolve, reject) => {
+      const options = {
+        flickrConsumerKey: this.apiKey,
+        flickrConsumerKeySecret: this.apiSecret,
+        oauthToken,
+        oauthTokenSecret,
+        oauthVerifier,
+        callback: (err, data) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve([data.oauthToken, data.oauthTokenSecret]);
+          }
+        },
+      };
+      flickrAPI.useRequestTokenToGetAccessToken(options);
+    });
+  }
+
+  callFlickr(method: string, params: ?{[key: string]: string}) {
+    return new Promise((res, rej) => {
+      flickrAPI.callApiMethod({
+        method: method,
+        flickrConsumerKey: this.apiKey,
+        flickrConsumerKeySecret: this.apiSecret,
+        oauthToken: this.accessToken,
+        oauthTokenSecret: this.accessTokenSecret,
+        optionalArgs: params,
+        callback: (err, data) => {
+          if (err) {
+            rej(err);
+          } else {
+            if (data.stat === 'fail') {
+              rej(new Error(data.message));
+            } else {
+              res(data);
+            }
+          }
+        },
+      });
+    });
+  }
+
+  async getPhotoset(id: string): Promise<Struct> {
```

Comment:
Yes. I thought about that on my to the office. flickr.js should not have any deps on noms.